src/components/PromotionList.tsx

```diff
@@ -105,28 +105,91 @@ export default function PromotionList(props: Props): ReactElement {
 
   /**
    * If the user has not saved the promotion, save the promotion. Otherwise, delete it from their saved promotions.
+   *
+   * @param promotionId - The id of the promotion to save
    */
   const onSaveButtonClick = useCallback(
     (promotionId: string) => {
-      if (!authUser) {
-        return Promise.reject(new Error('No user is logged in.'));
-      }
       const promos = [...promotions];
       const promotion = promos.find(({ id }) => id === promotionId);
       if (promotion) {
         if (promotion.isSavedByUser) {
           promotion.isSavedByUser = false;
-          return UserService.unsavePromotion(authUser.user.id, promotionId)
+          return UserService.unsavePromotion(authUser?.user.id || '', promotionId)
             .then(() => setPromotions(promos))
             .catch(() => null);
         }
         promotion.isSavedByUser = true;
-        return UserService.savePromotion(authUser.user.id, promotionId)
+        return UserService.savePromotion(authUser?.user.id || '', promotionId)
           .then(() => setPromotions(promos))
           .catch(() => null);
       }
     },
-    [promotions, setPromotions, authUser]
+    [authUser, promotions, setPromotions]
+  );
+
+  /**
+   * If the user has not downvoted the promotion, downvote the promotion. Otherwise, set back to initial state.
+   *
+   * @param promotionId - The id of the promotion to downvote
+   */
+  const onDownvoteClick = useCallback(
+    (promotionId: string): void => {
+      const promos = [...promotions];
+      const promotion = promos.find(({ id }) => id === promotionId);
+      if (promotion) {
+        if (promotion.voteState === VoteState.DOWN) {
+          promotion.votes = promotion.votes + 1;
+          promotion.voteState = VoteState.INIT;
+          PromotionService.downvotePromotion(promotionId, authUser?.user.id || '')
+            .then(() => setPromotions(promos))
+            .catch(() => null);
+        } else {
+          if (promotion.voteState === VoteState.UP) {
+            // Remove upvote
+            promotion.votes = promotion.votes - 1;
+          }
+          promotion.votes = promotion.votes - 1;
+          promotion.voteState = VoteState.DOWN;
+          PromotionService.downvotePromotion(promotionId, authUser?.user.id || '')
+            .then(() => setPromotions(promos))
+            .catch(() => null);
+        }
+      }
+    },
+    [authUser, promotions, setPromotions]
+  );
```

Comment:
What do you think about this?

Less duplication and same comment about defaulting userId to empty stirng. We can also try this for upvote

```suggestion
  const onDownvoteClick = useCallback(
    (promotionId: string) => {
      if (!authUser?.user?.id) {
        return Promise.reject(new Error('No user is logged in.'));
        // or do what richard suggested to navigate to login
      }
      const promos = [...promotions];
      const promotion = promos.find(({ id }) => id === promotionId);
      if (promotion) {
        if (promotion.voteState === VoteState.DOWN) {
          promotion.votes = promotion.votes + 1;
          promotion.voteState = VoteState.INIT;
        } else {
          if (promotion.voteState === VoteState.UP) {
            // Remove upvote
            promotion.votes = promotion.votes - 1;
          }
          promotion.votes = promotion.votes - 1;
          promotion.voteState = VoteState.DOWN;
        }

        PromotionService.downvotePromotion(promotionId, authUser.user.id)
          .then(() => setPromotions(promos))
          .catch(() => null);
      }
    },
    [authUser, promotions, setPromotions]
  );
```